tests/llama_index/test_llama_index_pyfunc_wrapper.py

```diff
@@ -0,0 +1,223 @@
+import pandas as pd
+import pytest
+from llama_index.core import QueryBundle
+from llama_index.core.base.response.schema import (
+    NodeWithScore,
+    Response,
+)
+from llama_index.core.chat_engine.types import (
+    AgentChatResponse,
+)
+from llama_index.core.llms import ChatMessage
+
+from mlflow.llama_index.pyfunc_wrapper import (
+    _CHAT_MESSAGE_HISTORY_PARAMETER_NAMES,
+    _CHAT_MESSAGE_PARAMETER_NAMES,
+    CHAT_ENGINE_NAME,
+    QUERY_ENGINE_NAME,
+    RETRIEVER_ENGINE_NAME,
+    SUPPORTED_ENGINES,
+    create_engine_wrapper,
+)
+
+
+@pytest.mark.parametrize("engine_type", SUPPORTED_ENGINES)
+def test_create_create_engine_wrapper(single_index, engine_type):
+    wrapped_model = create_engine_wrapper(single_index, engine_type)
+    assert wrapped_model is not None
+    assert wrapped_model.engine_type == engine_type
+    assert engine_type in str(wrapped_model.predict_callable)
+
+
+################## Inferece Input #################
+def test_format_predict_input_str_chat(single_index):
+    wrapped_model = create_engine_wrapper(single_index, CHAT_ENGINE_NAME)
+    formatted_data = wrapped_model._format_predict_input("string")
+    assert isinstance(formatted_data, str)
```

Comment:
```suggestion
    assert formatted_data == "string"
```

Let's strictly assert the content here and other tests, to avoid uncaught modification to the data.